docs/source/user-guide/transformations/joins.md

```diff
@@ -1,229 +1,268 @@
 # Joins
 
-## Join strategies
+A join is a dataframe operation in which the rows of two dataframes are concatenated horizontally according to a “joining strategy” and matching criteria.
```

Comment:
Though semantically correct, physically we don't concatenate rows and I don't want users to think in rows with Polars. Maybe we can find a different description here.

Which might also be a bit more correct as we may drop columns in a join. Inspired from wikipedia: 


"A join operation in Polars combines columns from one or more dataframes into a new dataframe"

Reply:
In hindsight, I don't love neither my original description nor your suggestion because we don't talk about the fact that usually something also happens to the rows.
(I do get your point about mentioning concatenation.)
E.g., anti and semi joins are all about the rows and 0% about the columns.

How about

 > A join operation combines columns from one or more dataframes into a new dataframe.
The different “joining strategies” and matching criteria used by the different types of joins influence how columns are combined and also what rows are included in the result of the join operation.